homeassistant/components/amberelectric/binary_sensor.py

```diff
@@ -71,6 +73,25 @@ def extra_state_attributes(self) -> dict[str, Any]:
         }
 
 
+class AmberDemandWindowBinarySensor(AmberPriceGridSensor):
+    """Sensor to show whether demand window is active."""
+
+    @property
+    def icon(self) -> str:
+        """Return the sensor icon."""
+        if self.is_on:
+            return DEMAND_WINDOW_ICONS[True]
+        return DEMAND_WINDOW_ICONS[False]
```

Comment:
Please use icon translations instead, for more information see:

https://developers.home-assistant.io/blog/2024/01/19/icon-translations/

Reply:
This is to be consistent with the other existing map in the same file. I suggest that we keep this PR as-is, and I can raise a separate PR to fix both. What do you think?

Reply:
I think we should not introduce new instances of patterns we don't want in our codebase anymore.

Please put this is an icon translations files, we can migrate others in follow up PRs.

Reply:
@frenck Could you suggest how I can test the icons when using icon translations?

Reply:
We do not test the working of icon translations itself, that specific codebase has been tested already in core; so no need to add extra tests for those.

Reply:
It's not that I don't trust the existing mechanism... it's that I'm not sure whether I'm implementing it in the right way. This sensor is time-based from a remote value, so it's pretty hard to check manually.

Reply:
Well, you are testing the state of the entity in that case (icons are a result of that).

Reply:
What I'm currently doing is:
```json
    "binary_sensor": {
      "demand_window": {
        "default": "mdi:meter-electric",
        "state": {
          "off": "mdi:meter-electric-outline"
        }
      }
    }
```

Is that the right way to do?

Specifically, I'm not sure what the state keys the binary sensor would use. I suppose it's `on` and `off` here but it's not all that obvious and as I mentioned I couldn't confirm manually.

Reply:
> Specifically, I'm not sure what the state keys the binary sensor would use.

Well, you test them:

![CleanShot 2024-07-06 at 11 51 02@2x](https://github.com/home-assistant/core/assets/195327/069975e0-1406-4855-9b66-da1b59b79710)

So yes, it is `on` and `off` :)